### PR TITLE
Add `pytorch/inference/{gpu,cpu}/2.3.1/transformers/4.46.0/py311`

### PR DESCRIPTION
**Pull request by alvarobartt**
## Description

This PR releases the new Hugging Face PyTorch DLC for Inference with a bunch of version dumps for `transformers`, `diffusers` and `sentence-transformers; as well as `torch` and other dependencies that otherwise could be problematic as the latest `huggingface_hub` release affecting `diffusers` <= 0.30.0

Some of the changes in this PR include:

- Include missing `requirements.txt` installation in `entrypoint.sh` (required to install custom dependencies with custom models).
- Fix `uvicorn` trigger on the `entrypoint.sh` to use `exec` so that the process runs with PID 1, allowing it to receive signals directly; and so on, to be gracefully shut down.
- Fix Python 3.11 installation as it was not properly installed and Python 3.10 was used instead; also fix `pip` installation to install via [`get-pip.py`](https://pip.pypa.io/en/stable/installation/#get-pip-py).
- Tested with some of the latest models that those bumped dependencies support as Gemma2, Llama3.2, StableDiffusion 3.5, and much more.

### CODE REVIEW

File: containers/pytorch/inference/cpu/2.3.1/transformers/4.46.0/py311/Dockerfile
Comment (philschmid) [inline containers/pytorch/inference/cpu/2.3.1/transformers/4.46.0/py311/Dockerfile:1]: should be 4.46.1 
image

File: containers/pytorch/inference/cpu/2.3.1/transformers/4.46.0/py311/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list && \
+    curl https://packages.cloud.google.com/apt/doc/apt-key.gpg \
+    | apt-key --keyring /usr/share/keyrings/cloud.google.gpg add - && \
+    apt-get update -y && \
+    apt-get install google-cloud-sdk -y && \
+    apt-get clean autoremove --yes && \
+    rm -rf /var/lib/{apt,dpkg,cache,log}
+
+# Copy entrypoint and change permissions
+COPY --chmod=0755 containers/pytorch/inference/cpu/2.3.1/transformers/4.46.0/py311/entrypoint.sh entrypoint.sh
```
Comment (philschmid) [inline containers/pytorch/inference/cpu/2.3.1/transformers/4.46.0/py311/Dockerfile:60]:
```suggestion
COPY --chmod=0755 containers/pytorch/inference/cpu/2.3.1/transformers/4.46.1/py311/entrypoint.sh entrypoint.sh
```

File: containers/pytorch/inference/cpu/2.3.1/transformers/4.46.0/py311/entrypoint.sh
Comment (philschmid) [inline containers/pytorch/inference/cpu/2.3.1/transformers/4.46.0/py311/entrypoint.sh:1]: same here

File: containers/pytorch/inference/gpu/2.3.1/transformers/4.46.0/py311/Dockerfile
Comment (philschmid) [inline containers/pytorch/inference/gpu/2.3.1/transformers/4.46.0/py311/Dockerfile:1]: same here

File: containers/pytorch/inference/gpu/2.3.1/transformers/4.46.0/py311/entrypoint.sh
Comment (philschmid) [inline containers/pytorch/inference/gpu/2.3.1/transformers/4.46.0/py311/entrypoint.sh:1]: same here

**Review by philschmid — APPROVED** [review]
LGTM except 4.46.0 is yanked and its 4.46.1

**Comment by alvarobartt** [comment]
> LGTM except 4.46.0 is yanked and its 4.46.1

Nice catch @philschmid thanks 🤗

File: containers/pytorch/inference/cpu/2.3.1/transformers/4.46.0/py311/Dockerfile
Comment (alvarobartt) [reply containers/pytorch/inference/cpu/2.3.1/transformers/4.46.0/py311/Dockerfile:1]: I'll create a quick patch PR on `huggingface-inference-toolkit` too.

**Pull request merged**